diy_view/src/main/java/scut/carsonho/diyview/SuperEditText.java

```diff
@@ -0,0 +1,422 @@
+package scut.carsonho.diyview;
+
+import ohos.agp.colors.RgbColor;
+import ohos.agp.components.AttrSet;
+import ohos.agp.components.Component;
+import ohos.agp.components.Component.DrawTask;
+import ohos.agp.components.Component.FocusChangedListener;
+import ohos.agp.components.Component.TouchEventListener;
+import ohos.agp.components.Text.TextObserver;
+import ohos.agp.components.TextField;
+import ohos.agp.components.element.Element;
+import ohos.agp.components.element.PixelMapElement;
+import ohos.agp.components.element.ShapeElement;
+import ohos.agp.render.Paint;
+import ohos.agp.utils.Color;
+import ohos.app.Context;
+import ohos.global.resource.NotExistException;
+import ohos.global.resource.Resource;
+import ohos.global.resource.ResourceManager;
+import ohos.hiviewdfx.HiLog;
+import ohos.hiviewdfx.HiLogLabel;
+import ohos.media.image.ImageSource;
+import ohos.media.image.PixelMap;
+import ohos.media.image.common.PixelFormat;
+import ohos.media.image.common.Rect;
+import ohos.media.image.common.Size;
+import ohos.multimodalinput.event.TouchEvent;
+import com.hmos.compat.utils.AttrUtils;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * Created by Carson_Ho on 17/8/4.
+ */
+public class SuperEditText extends TextField implements TextObserver,
+        FocusChangedListener, TouchEventListener, DrawTask {
+    private static final HiLogLabel HILOG_LABEL1 = new HiLogLabel(0, 0, "SuperEditText");
```

Comment:
Suffix "1" is not required for the variableName

Reply:
Done